opm/simulators/linalg/bda/rocm/rocsparseCPR.cpp

```diff
@@ -236,7 +236,9 @@ amg_cycle_gpu(const int level,
         HIP_CHECK(hipMemcpyAsync(h_y.data(), &y, sizeof(Scalar) * Ncur, hipMemcpyDeviceToHost, this->stream));
         
         // solve coarsest level using umfpack
-        this->umfpack.apply(h_x.data(), h_y.data());
+        if constexpr (std::is_same_v<Scalar,double>) {
```

Comment:
The AMG is only instansiated with double so I am not sure this is needed? If it is, when the `if` is evaluated to false, throw an error of some sort to signify that we cannot run apply?
And again, if this check is necessary, Include `<type_traits>` in this file because `is_same_v` is used

Reply:
the umfpack member is an int with float scalars, so we cannot de-reference thus compile fails without this. We will never actually hit this code however (it throws earlier), so no need for a warning here.

Reply:
Maybe add a comment explaining that this check is only there to make it compilable, not catch the error it looks like it is trying to catch

Reply:
done and done.